base/expr.jl

```diff
@@ -118,10 +118,42 @@ macro eval(x)
     :($(esc(:eval))($(Expr(:quote,x))))
 end
 
+"""
+    @inline
+
+Give a hint to the compiler that this function is worth inlining.
+
+Small functions typically don't even need the `@inline` annotation,
```

Comment:
Small nits on language. The `it` in `it does it on its own` does not refer to anything, should be spelled out `the compiler`. Also, needs a conjunction to join independent close (`as the compiler does`, `because the compiler`, `since the compilers`, etc). I'd remove the `even`, `don't even`, since it does not add anything to the documentation. I also prefer spelling out the `do not` in text such as documentation, but that's personal preference.